syntax = "proto3";

package input_provider;

// Input Provider Service - Provides initial configuration to the pipeline
service InputProvider {
  // Generate configuration (called by orchestrator via HTTP, result cached)
  rpc GetConfiguration(GetConfigurationRequest) returns (GetConfigurationResponse);

  // Fetch last cached result (called by downstream services via gRPC)
  rpc GetLastResult(Empty) returns (GetConfigurationResponse);
}

message Empty {}

message GetConfigurationRequest {
  // Empty request - configuration is predefined
}

message GetConfigurationResponse {
  bool success = 1;
  string message = 2;
  int32 num_records = 3;  // Number of records to generate
}
